vars/neoDeploy.groovy

```diff
@@ -19,33 +19,104 @@ def call(parameters = [:]) {
+            applicationName = utils.getMandatoryParameter(parameters, 'applicationName', null)
+            runtime = utils.getMandatoryParameter(parameters, 'runtime', null)
+            runtimeVersion = utils.getMandatoryParameter(parameters, 'runtimeVersion', null)
+            vmSize = utils.getMandatoryParameter(parameters, 'vmSize', 'lite')
+            if (vmSize != 'lite' && vmSize !='pro' && vmSize != 'prem' && vmSize != 'prem-plus') {
+                throw new IllegalArgumentException("[neoDeploy] Invalid vmSize = '${vmSize}'. Valid 'vmSize' values are: 'lite', 'pro', 'prem' and 'prem-plus'.")
+            }
+        }
+
         def defaultDeployHost = script.commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
         def defaultDeployAccount = script.commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
         def defaultCredentialsId = script.commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
         if (defaultCredentialsId == null) {
             defaultCredentialsId = 'CI_CREDENTIALS_ID'
         }
 
-        def deployHost = utils.getMandatoryParameter(parameters, 'deployHost', defaultDeployHost)
-        def deployAccount = utils.getMandatoryParameter(parameters, 'deployAccount', defaultDeployAccount)
+        def deployHost
+        def deployAccount
+
+        if (deployMode.equals('MTA') || deployMode.equals('WAR_PARAMS')) {
+            deployHost = utils.getMandatoryParameter(parameters, 'deployHost', defaultDeployHost)
+            deployAccount = utils.getMandatoryParameter(parameters, 'deployAccount', defaultDeployAccount)
+        }
+
         def credentialsId = parameters.get('neoCredentialsId', defaultCredentialsId)
 
         def neoExecutable = getNeoExecutable(parameters)
 
         withCredentials([usernamePassword(
-                credentialsId: credentialsId,
-                passwordVariable: 'password',
-                usernameVariable: 'username'
-        )]) {
-            sh """#!/bin/bash
-                    "${neoExecutable}" deploy-mta \
-                      --user '${username}' \
+            credentialsId: credentialsId,
+            passwordVariable: 'password',
+            usernameVariable: 'username')]) {
+
+            def commonDeployParams =
+                """--user '${username}' \
+                   --password '${password}' \
+                   --source "${archivePath.getAbsolutePath()}" \
+                """
+
+            if (deployMode == 'MTA') {
+                sh """#!/bin/bash
+                      "${neoExecutable}" deploy-mta \
+                      ${commonDeployParams} \
                       --host '${deployHost}' \
-                      --source "${archivePath.getAbsolutePath()}" \
                       --account '${deployAccount}' \
-                      --password '${password}' \
                       --synchronous
-               """
+                   """
+            }
+
+            if (deployMode == 'WAR_PARAMS') {
```

Comment:
Could you please convert the parameter values to lower case. Otherwise it would be different to other steps.